prover/witness_generator/src/metrics.rs

```diff
@@ -0,0 +1,65 @@
+use std::time::Duration;
+use vise::{
+    Buckets, Counter, EncodeLabelSet, EncodeLabelValue, Family, Gauge, Histogram, LabeledFamily,
+    Metrics,
+};
+use zksync_types::proofs::AggregationRound;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelValue, EncodeLabelSet)]
+#[metrics(label = "stage", format = "wit_gen_{}")]
+pub(crate) struct StageLabel(AggregationRound);
+
+impl From<AggregationRound> for StageLabel {
+    fn from(round: AggregationRound) -> Self {
+        Self(round)
+    }
+}
+
+impl std::fmt::Display for StageLabel {
+    fn fmt(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        formatter.write_str(match self.0 {
+            AggregationRound::BasicCircuits => "basic_circuits",
+            AggregationRound::LeafAggregation => "leaf_aggregation",
+            AggregationRound::NodeAggregation => "node_aggregation",
+            AggregationRound::Scheduler => "scheduler",
```

Comment:
wait, didn't you have this metrics above? I'm confused now

Reply:
Do you mean this StageLabel type? If yes, then the thing is that we need to use AggregationRound, which is an external type and we use wrappers for it here. Since all these metrics are in different crates, I wasn't really sure how to work with them.